packages/grid/_modules_/grid/components/columnHeaders/ColumnHeaderSeparator.tsx

```diff
@@ -15,10 +14,9 @@ export const ColumnHeaderSeparator = React.memo(function ColumnHeaderSeparator(
   props: ColumnHeaderSeparatorProps,
 ) {
   const { resizable, resizing, height, ...other } = props;
-  const icons = useIcons();
   const apiRef = React.useContext(ApiContext);
   const { showColumnRightBorder } = useGridSelector(apiRef, optionsSelector);
-  const Icon = icons!.ColumnResize!;
+  const ColumnResizeIcon = apiRef!.current.components!.ColumnResizeIcon!;
```

Comment:
Just a general thought: what do you think of having the icons nested in the `components` like `components.icons` to keep them grouped? If it is too much work then just ignore this comment :).

Reply:
We talked about it beforehand and went against the idea because 1. the flattened the structure is, the easier to navigate it comes. 2. there is nothing special about the icons, they are like any other components. We will offer the capability to customize the Checkbox, Button, everything long-term.

Reply:
Ok, sorry, I must have missed that.

Reply:
We didn't share this design decision inside the related issue (#840), it was on a private communication channel. I wish we stick to a written channel next tiem.